staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -190,6 +190,18 @@ func (s *GenericAPIServer) PrepareRun() preparedGenericAPIServer {
 // Run spawns the http servers (secure and insecure). It only returns if stopCh is closed
 // or one of the ports cannot be listened on initially.
 func (s preparedGenericAPIServer) Run(stopCh <-chan struct{}) error {
+	err := s.NonBlockingRun(stopCh)
+	if err != nil {
+		return err
+	}
+
+	<-stopCh
+	return nil
+}
+
+// NonBlockingRun spawns the http servers (secure and insecure). An
+// error is returned if one of the ports cannot be listened on.
```

Comment:
I think for now docs will have to be enough. The behavior of the blocking call doesn't change (callers are intended to exit on failure), and users of the non-blocking call (i.e. tests) will need to close the channel to avoid leaking resources.